os_migrate/plugins/module_utils/server_port.py

```diff
@@ -78,7 +80,12 @@ def from_sdk(cls, conn, sdk_resource):
         return obj
 
     def create_or_update(self, conn, filters=None):
-        raise exc.Unsupported("Direct ServerPort.create_or_update call is unsupported.")
+        refs = self._refs_from_ser(conn)
+        sdk_params = self._to_sdk_params(refs)
+
+        # creation using neutron, NOTE: add update method
+        sdk_port = conn.network.create_port(**sdk_params)
+        return {'id': sdk_port['id']}
```

Comment:
Why 'id'? See the doc on the 'networks' parameter how to pass in a port reference: https://docs.openstack.org/api-ref/compute/?expanded=create-server-detail#create-server 

Perhaps we should just return `sdk_port` from here and do the parameter molding entirely in the `update_sdk_params_networks_simple`, that would be cleaner and less confusing approach perhaps. It would make it more visible how the `networks` parameter is being formed.

Reply:
networks | body | array | A list of network object. Required parameter when there are multiple networks defined for the tenant. When you do not specify the networks parameter, the server attaches to the only network created for the current tenant.  Optionally, you can create one or more NICs on the server. To provision the server instance with a NIC for a network, specify the UUID of the network in the uuid attribute in a networks object.  To provision the server instance with a NIC for an already existing port, specify the port-id in the port attribute in a networks object. ...
-- | -- | -- | --

...



networks.port (Optional) | body | string | To provision the server instance with a NIC for an already existing port, specify the port-id in the port attribute in a networks object. The port status must be DOWN. Required if you omit the uuid attribute. Requested security groups are not applied to pre-existing ports.
-- | -- | -- | --